presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/ColumnStatistics.java

```diff
@@ -231,13 +216,19 @@ public String toString()
     @Override
     public void addHash(StatisticsHasher hasher)
     {
+        // This hashing function is used for ORC writer file validation.
+        // Fields rawSize and storageSize are not included because they cannot
+        // be calculated by the reader during the file validation.
         hasher.putOptionalLong(hasNumberOfValues, numberOfValues)
-                .putOptionalLong(hasRawSize, rawSize)
-                .putOptionalLong(hasStorageSize, storageSize)
                 .putOptionalHashable(getBloomFilter());
     }
 
     public static ColumnStatistics mergeColumnStatistics(List<ColumnStatistics> stats)
+    {
+        return mergeColumnStatistics(stats, null);
+    }
+
+    public static ColumnStatistics mergeColumnStatistics(List<ColumnStatistics> stats, Long extraStorageSize)
```

Comment:
Storage size can be computed only for the file stats after flushing all stripes. But the column stats are immutable and are already provided by the column writers. That's why I decided to pass the storage size into a function that merges stripe column stats into a file column stat.